app/gui/src/controller/searcher.rs

```diff
@@ -1043,8 +1041,22 @@ impl Searcher {
     fn make_component_list<'a>(
         &self,
         entry_ids: impl IntoIterator<Item = suggestion_database::entry::Id>,
+        this_type: &Option<String>,
+        return_types: &[String],
     ) -> component::List {
         let mut builder = self.list_builder_with_favorites.deref().clone();
+        if this_type.is_none() {
+            let snippets = if return_types.is_empty() {
+                component::hardcoded::INPUT_SNIPPETS.with(|s| s.clone())
+            } else {
+                let parse_type_qn = |s| tp::QualifiedName::from_text(s).ok();
+                let rt_qns = return_types.iter().filter_map(parse_type_qn);
+                component::hardcoded::input_snippets_with_matching_return_type(rt_qns)
+            };
+            let group_name = component::hardcoded::INPUT_GROUP_NAME;
+            let project = project::QualifiedName::standard_base_library();
+            builder.insert_virtual_components_in_favorites_group(group_name, project, snippets);
+        }
```

Comment:
I think it could be extracted to a special function "add_virtual_entries_to_builder", so it'll be clear what this block is for.